source/bringauto/common_utils/ProtobufUtils.cpp

```diff
@@ -12,36 +8,33 @@ InternalProtocol::InternalServer
 ProtobufUtils::createInternalServerConnectResponseMessage(const InternalProtocol::Device &device,
 														  const InternalProtocol::DeviceConnectResponse_ResponseType &resType) {
 	InternalProtocol::InternalServer message {};
-	auto response = message.mutable_deviceconnectresponse();
+	const auto response = message.mutable_deviceconnectresponse();
 	response->set_responsetype(resType);
-	auto device_ = response->mutable_device();
-	device_->CopyFrom(device);
+	*response->mutable_device() = std::move(device);
```

Comment:
Device is const and you move it. So  it completely breaks semantics and the function itself can be used wrongly. Do not MOVE const references!